### PR TITLE
feat: Send connection issues to analytics

### PR DESCRIPTION
**Pull request by irmatov**
Closes https://github.com/cloudquery/cloudquery-issues/issues/370

### CODE REVIEW

File: cmd/fetch/fetch.go
Comment (roneli) [inline cmd/fetch/fetch.go:67]: I can see this growing into many event types, we currently have fetch and policy, but in this case we can have more.
Maybe call this event type "diag" and give it sub types and some meta information as occurrence count etc' and by no means to pass the "error" itself as that can have user information we don't want.

ill let @bbernays  weigh in some more here.

**Review by roneli — CHANGES_REQUESTED** [review]
Nice idea for adding more meta analytic information, we do need to update the docs after this PR merges. Added some thoughts here.

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Let's hold on with that as it uses Diags which are going away in the upcoming weeks - https://github.com/cloudquery/cq-provider-sdk/pull/437

**Comment by yevgenypats** [comment]
Closing as we move to cloudquery v2 and error reporting is different

**Pull request closed without merging**
